meta/TreeMasses.m

```diff
@@ -408,6 +409,20 @@
 
 IsElectricallyCharged[par_] := GetElectricCharge[par] != 0;
 
+IsChargedUnder[field_, vector_] := Block[{},
+  (*First of all, make sure we have been passed a vector*)
+  If[IsVector[vector],
+    (*Check 2 special cases first which are quicker*)
+    Which[IsPhoton[vector], Return[IsElectricallyCharged[field]];,
+     IsGluon[vector], Return[ColorChargedQ[field]];,
+     (*Else check that this field coupled with its anti-field can emit this vector*)
+     (*Note this will not work for vectors that couple to two different fields, e.g. W-bosons*)
+     True, Return[-I SARAH`Vertex[{SARAH`AntiField[field], field, 
```

Comment:
If we're just checking if the vertex is != 0, the `-I` maybe is an overkill :wink:

Reply:
Good point, this is a leftover from the borrowed code from GetElectricCharge[].  Thanks.